src/lammpsio/snapshot.py

```diff
@@ -84,14 +84,20 @@ def from_hoomd_gsd(cls, frame):
             box=box,
             step=frame.configuration.step,
         )
-        snap.position = frame.particles.position
-        snap.velocity = frame.particles.velocity
-        snap.image = frame.particles.image
-        snap.typeid = frame.particles.typeid + 1
-        snap.charge = frame.particles.charge
-        snap.mass = frame.particles.mass
-
-        snap.molecule = frame.particles.body + 1
+        if frame.particles.position is not None:
+            snap.position = frame.particles.position
+        if frame.particles.velocity is not None:
+            snap.velocity = frame.particles.velocity
+        if frame.particles.image is not None:
+            snap.image = frame.particles.image
+        if frame.particles.typeid is not None:
+            snap.typeid = frame.particles.typeid + 1
+        if frame.particles.charge is not None:
+            snap.charge = frame.particles.charge
+        if frame.particles.mass is not None:
+            snap.mass = frame.particles.mass
+        if frame.particles.body is not None:
+            snap.molecule = frame.particles.body + 1
         if numpy.any(snap.molecule < 0):
             warnings.warn("Some molecule IDs are negative, remapping needed.")
```

Comment:
Move this inside the `if body is not None` check, it only needs to run if this field gets set